OS detection fails in macOS

OS detection fails to detect macOS.
In macOS 15.2, OpenJDK 21:

```
Caused by: java.lang.IllegalArgumentException: Unsupported OS: mac os x
        at org.opentorah.platform.Os$.$anonfun$2(Os.scala:22)
        at scala.Option.getOrElse(Option.scala:201)
        at org.opentorah.platform.Os$.get(Os.scala:22)
        at org.opentorah.node.NodeInstallation$.fromOs(NodeInstallation.scala:25)
        at org.opentorah.node.NodeExtension.node(NodeExtension.scala:25)
        at org.podval.tools.scalajs.ScalaJSPlugin.apply$$anonfun$1(ScalaJSPlugin.scala:37)
```

```
$ scala-cli repl
Welcome to Scala 3.5.2 (21.0.5, Java OpenJDK 64-Bit Server VM).
Type in expressions for evaluation. Or try :help.
                                                                                                                                                                                                                                                   
scala> System.getProperty("os.name")
val res0: String = Mac OS X
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -6,6 +6,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [0.10.14] - 2025-01-07
+- chore: dependency updates
+- fix: OS detection fails in macOS  https://github.com/opentorah/opentorah/issues/341
+
 ## [0.10.13] - 2024-12-22
 - chore: Gradle 8.12 and related updates
 - chore: dependency updates

--- a/build.gradle
+++ b/build.gradle
@@ -19,9 +19,3 @@ idea {
 
 // There is no Java in the project :)
 project.gradle.startParameter.excludedTaskNames.add('compileJava')
-
-// TODO eliminate the use of `subprojects` - move this into `configure.gradle`?
-subprojects {
-  group = 'org.opentorah'
-  version = '0.10.13'
-}

--- a/build.gradle
+++ b/build.gradle
@@ -20,6 +20,8 @@ idea {
 // There is no Java in the project :)
 project.gradle.startParameter.excludedTaskNames.add('compileJava')
 
-// TODO is this needed?
-group = 'org.opentorah'
-version = '0.10.13'
+// TODO eliminate the use of `subprojects` - move this into `configure.gradle`?
+subprojects {
+  group = 'org.opentorah'
+  version = '0.10.13'
+}

--- a/configure.gradle
+++ b/configure.gradle
@@ -1,4 +1,6 @@
-// Scala
+project.version = '0.10.14'
+project.group = 'org.opentorah'
+
 dependencies {
   implementation libs.scala.library
 
@@ -44,7 +46,7 @@ test {
 
 // library: all projects are configured as a library (except for the docs one);
 // with the latest Gradle plugin publishing plugin, uploading of it as a library started failing, so I disabled it...
-if (project.name != 'docs' && project.name != 'opentorah-docbook') {
+if (project.name != 'docs') {
   final String gitHubRepository = "opentorah/opentorah"
   final String gitHubRepositoryUrl = "https://github.com/$gitHubRepository"
   final String orgName = 'Open Torah Project'

--- a/settings.gradle
+++ b/settings.gradle
@@ -6,7 +6,7 @@ pluginManagement {
 
   plugins {
     id 'com.github.ben-manes.versions' version '0.44.0'
-    id 'org.podval.tools.scalajs'      version '0.4.12'
+    id 'org.podval.tools.scalajs'      version '0.4.13'
     id 'com.google.cloud.tools.jib'    version '3.4.4'
     id 'org.podval.tools.cloudrun'     version '0.4.0'
     id 'com.gradle.plugin-publish'     version '1.2.1'

--- a/settings.gradle
+++ b/settings.gradle
@@ -36,7 +36,7 @@ dependencyResolutionManagement {
     // Scala
     library('scala-library', "org.scala-lang:scala3-library_3:$scalaVersion")
     library('scala-xml'    , "org.scala-lang.modules:scala-xml_$scalaVersionMajor:2.3.0")
-    library('zinc'         , 'org.scala-sbt:zinc_2.13:1.10.5')
+    library('zinc'         , 'org.scala-sbt:zinc_2.13:1.10.7')
 
     // XML
     library('xerces'       , 'xerces:xercesImpl:2.12.2')
@@ -68,7 +68,7 @@ dependencyResolutionManagement {
 
     // Logging
     library('slf4j'   , 'org.slf4j:slf4j-api:2.0.16')
-    library('logback' , 'ch.qos.logback:logback-classic:1.5.15')
+    library('logback' , 'ch.qos.logback:logback-classic:1.5.16')
     library('logstash', 'net.logstash.logback:logstash-logback-encoder:8.0')
 
     // Flexmark

--- a/util/build.gradle
+++ b/util/build.gradle
@@ -3,7 +3,9 @@ plugins {
   id 'scala'
   id 'maven-publish'
   id 'signing'
-  id 'org.podval.tools.scalajs'
+  // TODO this still breaks *sometimes*, but I do not want to use maiflai plugin any more
+  // (also, it is not compatible with Gradle 8.11...)
+//  id 'org.podval.tools.scalajs'
 }
 
 description = 'Open Torah Util'

--- a/util/src/main/scala/org/opentorah/platform/Os.scala
+++ b/util/src/main/scala/org/opentorah/platform/Os.scala
@@ -1,24 +1,44 @@
 package org.opentorah.platform
 
-enum Os(
-  val hasUname: Boolean = false,
-  val libraryExtension: String = "so"
-) derives CanEqual:
-  case Windows extends Os(libraryExtension = "dll")
-  case Aix
-  case Mac extends Os(libraryExtension = "dylib")
-  case FreeBSD
-  case SunOS
-  case Linux extends Os(hasUname = true)
-  case Android
+trait Os derives CanEqual:
+  def hasUname: Boolean = false
+  def libraryExtension: String = "so"
+  def is(name: String): Boolean
 
 object Os:
+  case object Android extends Os:
+    override def is(name: String): Boolean = 
+      Linux.is(name) &&
+      System.getProperty("java.specification.vendor").contains("Android")
+
+  case object Linux extends Os:
+    override def is(name: String): Boolean = name.startsWith("Lin")
+    override def hasUname: Boolean = true
+
+  case object Mac extends Os:
+    override def is(name: String): Boolean = name.startsWith("Mac")
+    override def libraryExtension: String = "dylib"
+
+  case object Windows extends Os:
+    override def is(name: String): Boolean = name.startsWith("Win")
+    override def libraryExtension: String = "dll"
+
+  case object Aix extends Os:
+    override def is(name: String): Boolean = name.startsWith("Aix")
+
+  case object FreeBSD extends Os:
+    override def is(name: String): Boolean = name.startsWith("Free")
+
+  case object SunOS extends Os:
+    override def is(name: String): Boolean = name.startsWith("Sun")
+
+  // Android is always Linux and thus has to be earlier in the list:
+  val values: Seq[Os] = Seq(Android, Linux, Mac, Windows, Aix, FreeBSD, SunOS)
+
   def getName: String = System.getProperty("os.name")
 
-  def get: Os =
-    val name: String = getName.toLowerCase
-    val result = Os.values
-      .find(_.toString.toLowerCase.contains(name.toLowerCase))
-      .getOrElse(throw IllegalArgumentException(s"Unsupported OS: $name"))
-    if result == Os.Linux && System.getProperty("java.specification.vendor").contains("Android") then Os.Android
-    else result
+  def get: Os = get(getName)
+
+  def get(name: String): Os = values
+    .find(_.is(name))
+    .getOrElse(throw IllegalArgumentException(s"Unsupported Operating System: $name"))

